### PR TITLE
Channel test return error response

### PR DESCRIPTION
**Pull request by qnkhuat**
Follow up of https://github.com/metabase/metabase/pull/49661, we want to surface the response if the test request fails.

See this test for how the response would look like https://github.com/metabase/metabase/pull/50254/files#diff-55c06ecf148871a20347412ddf41f2754a028295033cccd0e71ae040c482a167R119

### CODE REVIEW

File: src/metabase/channel/impl/http.clj [hunk 1]
```diff
@@ -60,6 +60,7 @@
     true
     (catch Exception e
       (let [data (ex-data e)]
+        (def data data)
```
Comment (piranha) [inline src/metabase/channel/impl/http.clj:63]: def data data :)

**Review by sansolovyov — APPROVED** [review]

File: test/metabase/channel/impl/http_test.clj [hunk 2]
```diff
@@ -54,8 +55,11 @@
 
 (defn do-with-server
   [route+handlers f]
+  (def route+handlers route+handlers)
   (let [handler        (->> route+handlers
-                            (map :route)
+                            (mapv :route)
+                            (cons (compojure.route/not-found {:status-code 404 :body "Not found."}))
+                            reverse ;; not found is last
```
Comment (crisptrutski) [inline test/metabase/channel/impl/http_test.clj:62]: It was already a vector, why not use `conj` instead of `cons` to add it to the other side?

Comment (qnkhuat) [reply test/metabase/channel/impl/http_test.clj:62]: bc this is a thread last 😞

Comment (crisptrutski) [reply test/metabase/channel/impl/http_test.clj:61]: How about using as->?

**Pull request merged**